src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -2093,6 +2094,106 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         }
     }
 
+    void ControlCore::SelectCommand(const bool goUp)
+    {
+        const til::point start = HasSelection() ? (goUp ? _terminal->GetSelectionAnchor() : _terminal->GetSelectionEnd()) :
+                                                  _terminal->GetTextBuffer().GetCursor().GetPosition();
+        std::optional<DispatchTypes::ScrollMark> nearest{ std::nullopt };
+        const auto marks{ std::ranges::reverse_view(_terminal->GetScrollMarks()) };
+        for (auto&& m : marks)
+        {
+            // If this mark doesn't know anything about the position of its
+            // command, OR it does but thinks that it was empty, then just skip
+            // it.
+            if (!m.commandEnd.has_value() || *m.commandEnd == m.end)

```

Comment:
This condition makes me think that `ScrollMark` could use some small helper member functions. If I'm reading this right, this would be something like `!m.HasCommand()`? Because with that in place we can also drop the comment here.